src/utils/relativeTime.js

```diff
@@ -0,0 +1,64 @@
+import i18n from '@dhis2/d2-i18n'
+
+const language = i18n.language?.replace('_', '-') || 'en-gb'
+
+export const [selectedLocale] = Intl.DateTimeFormat.supportedLocalesOf([
+    language,
+    'en-gb',
+])
+
+/**
+ * Find which of the two locales is supported by the user's browser for relative time formatting
+ * */
+const [relativeTimeLocale] = Intl.RelativeTimeFormat.supportedLocalesOf([
+    language,
+    'en',
+])
+
+/**
+ * Time unit and its equivalent value in milliseconds
+ * */
+const timeUnits = [
+    { unit: 'year', ms: 31536000000 },
+    { unit: 'month', ms: 2628000000 },
+    { unit: 'day', ms: 86400000 },
+    { unit: 'hour', ms: 3600000 },
+    { unit: 'minute', ms: 60000 },
+    { unit: 'second', ms: 1000 },
+]
+
+const relativeTimeFormatter = new Intl.RelativeTimeFormat(relativeTimeLocale, {
+    numeric: 'always',
+})
+
+/**
+ * Finds the most appropriate time unit
+ * */
+const getClosestTimeUnit = (deltaMs, minUnit = 'millisecond') => {
+    const absoluteDelta = Math.abs(deltaMs)
+    for (const { unit, ms } of timeUnits) {
+        if (absoluteDelta >= ms || unit === minUnit) {
+            return { unit, value: Math.round(deltaMs / ms) }
+        }
+    }
+    return { unit: 'millisecond', value: deltaMs }
+}
+
+/**
+ * Get language sensitive relative time message from elapsed time
+ */
+const getRelativeTimeFromDelta = (deltaMs) => {
+    if (Math.abs(deltaMs) < 1000) {
+        return i18n.t('Now')
+    }
+    const { value, unit } = getClosestTimeUnit(deltaMs, 'second')
+    return relativeTimeFormatter.format(value, unit)
+}
+
+/**
+ * Get language sensitive relative time message from Dates
+ */
+export const getRelativeTime = (relative, baseDate = new Date()) => {
```

Comment:
cc @Birkbjo: 
this code looks good, and I don't like using moment, but in the past when I've asked about this, I feel like the conclusion was "let's just use moment.from() because we're already importing moment". 

I feel like if we're going to use our own relative time logic, we should put it in a common repo (e.g. multi-calendar-dates?) so that we can centralize the logic/testing. It seems inefficient/ultimately error-prone to have this code be copied into multiple repos. (also then seems to put a burden on retesting functionality in the app)

Reply:
Yeah, I agree we should try to centralize this.
But this is a fresh app, and it wouldn't be ideal to add moment as a dependency, right?